### ISSUE TITLE
[ENH] add temporal aggregation seriestoseries transformer

### ISSUE THREAD

**Issue by AngelPone** [sktime/sktime#1667]
**Is your feature request related to a problem? Please describe.**

Temporal aggregation aggregates high-frequency time series into lower frequency time series using a fixed nonoverlapping window. For example, we can aggregate daily time series into weekly time series using a window of length 7. Temporal aggregation in different levels can strengthen or attenuate different features of the time series, and thus reveal import  information about the data generating process.
Currently, there are similar forecasting strategies such as `makereduction` using a overlapping sliding-window, but there are not similar methods in transformer.

**Describe the solution you'd like**

Add temporal aggregation transformer. User can specify the window (e.g. `6` for monthly time series to produce half-year time series)

**Additional context**

Temporal aggregation has been explored widely in forecasting literatures. There are multiple forecasting methods(e.g. MAPA, iMAPA, ADIDA) that combine  forecasts from multiple aggregation levels, which improve forecasting accuracy and decrease uncertainty. But they cannot be implemented in sktime elegantly because aggregated series in different levels have different length, also see #230 and Unequal length data in roadmap  #228.

**Comment by khrapovs** [comment]
Probably I still do not entirely understand how sktime works. Yesterday I started implementing ResamplingTransformer but quickly deleted the branch when I got a very bad feeling that I am just writing a wrapper for `.resample()` method of pandas. The transformation in its core is just one line, e.g. `.resample("W").mean()`. Why should we build something restrictive when a user can him/herself create a transformer step with required frequency, aggregation function, other settings? Or is it not possible in the current version of sktime to create custom transformers and custom pipelines?

**Comment by aiwalter** [comment]
@khrapovs there is a `FunctionTrasnformer`, maybe you can check if that is what you are looking for?
https://github.com/alan-turing-institute/sktime/blob/5fecb32124d4776e542b6f9ddaddf5b3e7f3b680/sktime/transformations/series/func_transform.py

Seems its not yet added to docs.

**Comment by khrapovs** [comment]
@aiwalter Yes, for example. I saw this transformer and I agree, this works perfectly fine:
```
from sktime.datasets import load_airline
from sktime.transformations.series.func_transform import FunctionTransformer
y = load_airline()
transformer = FunctionTransformer(lambda s: s.resample("Y").mean())
transformer.fit_transform(y)
```

**Comment by fkiraly** [comment]
implemented in https://github.com/sktime/sktime/pull/3997
